crates/biome_project/src/node_js_project/package_json.rs

```diff
@@ -21,6 +24,34 @@ pub struct PackageJson {
     pub r#type: Option<PackageType>,
 }
 
+impl PackageJson {
+    /// Checks whether the `specifier` is defined in `dependencies`, `dev_dependencies` or `peer_dependencies`
+    pub fn contains_dependency(&self, specifier: &str) -> bool {
+        self.dependencies.contains(specifier)
+            || self.dev_dependencies.contains(specifier)
+            || self.peer_dependencies.contains(specifier)
+    }
+
+    /// Checks whether the `specifier` is defined in `dependencies`, `dev_dependencies` or `peer_dependencies`, and the `range`
+    /// of matches the one of the manifest
+    pub fn matches_dependency(&self, specifier: &str, range: &str) -> bool {
+        let iter = self
+            .dependencies
+            .iter()
+            .chain(self.dev_dependencies.iter())
+            .chain(self.peer_dependencies.iter());
+        for (dependency_name, dependency_version) in iter {
+            if let Ok(range) = Range::from_str(range) {
```

Comment:
I think we could move the conversion out of the loop.
```suggestion
        let Ok(range) = Range::from_str(range) else { return false };
        for (dependency_name, dependency_version) in iter {
```
Another possibility is to parse the range only when a dependency name matches.